### PR TITLE
dir: Export d-bus .service files for names that are are allowed to own

### PR DESCRIPTION
**Pull request by smspillaz**
Not just ones that match our app ID.

### CODE REVIEW

**Comment by alexlarsson** [comment]
This still seems to not have the last fixes.

**Comment by alexlarsson** [comment]
Hmm, bunch of test failures here...

**Comment by smspillaz** [comment]
> Hmm, bunch of test failures here...

Hmm, okay, so the weirdness I was seeing was not a localized failure then. I suspect the problem might be here: https://github.com/smspillaz/flatpak/blob/export-owned-dbus-names-2/common/flatpak-dir.c#L5084 (originally the second `name` was `g_file_get_path (source)`, but I renamed it to enable prefix matching to work).

**Comment by alexlarsson** [comment]
@rh-atomic-bot r+

**Comment by alexlarsson** [comment]
bot, add author to whitelist

File: common/flatpak-dir.c [hunk 1]
```diff
@@ -4888,15 +4888,56 @@ export_desktop_file (const char   *app,
...
+
+      for (; *iter != NULL; ++iter)
+        g_ptr_array_add (prefixes, g_strdup (*iter));
+
+      *out_match_prefixes_func = flatpak_name_matches_one_wildcard_prefix;
+    }
+
+  g_ptr_array_add (prefixes, NULL);
+
+  *out_match_prefixes_func = flatpak_name_matches_one_prefix;
```
Comment (ramcq) [inline common/flatpak-dir.c]: This assignment seems to unconditionally clobber the one a few lines up.

**Comment by ramcq** [comment]
@smspillaz @alexlarsson Looks like a little bug with prefix matching functions?

File: common/flatpak-dir.c (hunk 1 above)
Comment (smspillaz) [reply common/flatpak-dir.c]: Ah, yes, you're right. I'll follow up.

**Pull request closed without merging**